### PR TITLE
enh(UI): Create form to configure OpenID Connect

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This adds a form to configure OpenID Connect.

https://user-images.githubusercontent.com/12515407/155724256-e3683fea-8d83-49c8-baa6-f899d53860a7.mov

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Go to Administration > Authentication
- Click on "Open ID Configuration" tab
- -> A base Open ID configuration is displayed
- Change some values
- Click on the "Save" button
- -> Your configuration is correctly saved

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -16533,3 +16533,78 @@ msgstr "Erreur lors de la liaison du modèle (id: %d) à l'hôte (id: %d)"
...
+msgstr "Activer l'authentification OpenID Connect"
+
+msgid "Authentication mode"
+msgstr "Mode d'authentification"
+
+msgid "Trusted client addresses"
+msgstr "Adresses de clients de confiance"
+
+msgid "Blacklist client addresses"
+msgstr "Adresses de clients blacklistés"
```
Comment (jdelpierre) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16568]: Don't we say "liste noire"?

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -16533,3 +16533,78 @@ msgstr "Erreur lors de la liaison du modèle (id: %d) à l'hôte (id: %d)"
...
+msgid "OpenID Connect Configuration"
+msgstr "Configuration OpenID Connect"
+
+msgid "Define OpenID Connect Configuration"
+msgstr "Définir la configuration OpenID Connect"
+
+msgid "Invalid URL"
+msgstr "URL invalide"
+
+msgid "Invalid IP Address or Domain name"
```
Comment (jdelpierre) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16549]:
```suggestion
msgid "Invalid IP Address or domain name"
```

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -14890,3 +14890,78 @@ msgstr ""
...
+# msgid "OpenID Connect Configuration"
+# msgstr ""
+
+# msgid "Define OpenID Connect Configuration"
+# msgstr ""
+
+# msgid "Invalid URL"
+# msgstr ""
+
+# msgid "Invalid IP Address or Domain name"
```
Comment (jdelpierre) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po:14906]:
```suggestion
# msgid "Invalid IP Address or domain name"
```

File: www/front_src/src/Authentication/Openid/translatedLabels.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+export const labelOpenIDConnectConfiguration = 'OpenID Connect Configuration';
+export const labelDefineOpenIDConnectConfiguration =
+  'Define OpenID Connect Configuration';
+export const labelRequired = 'Required';
+export const labelInvalidURL = 'Invalid URL';
+export const labelInvalidIPAddressOrDomainName =
+  'Invalid IP Address or Domain name';
```
Comment (jdelpierre) [inline www/front_src/src/Authentication/Openid/translatedLabels.tsx:7]:
```suggestion
  'Invalid IP Address or domain name';
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -15341,3 +15341,78 @@ msgstr ""
...
+# msgid "OpenID Connect Configuration"
+# msgstr ""
+
+# msgid "Define OpenID Connect Configuration"
+# msgstr ""
+
+# msgid "Invalid URL"
+# msgstr ""
+
+# msgid "Invalid IP Address or Domain name"
```
Comment (jdelpierre) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:15357]:
```suggestion
# msgid "Invalid IP Address or domain name"
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -15329,3 +15329,78 @@ msgstr ""
...
+# msgid "OpenID Connect Configuration"
+# msgstr ""
+
+# msgid "Define OpenID Connect Configuration"
+# msgstr ""
+
+# msgid "Invalid URL"
+# msgstr ""
+
+# msgid "Invalid IP Address or Domain name"
```
Comment (jdelpierre) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:15345]:
```suggestion
# msgid "Invalid IP Address or domain name"
```

File: www/front_src/src/Authentication/Openid/useValidationSchema.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import {
+  labelRequired,
+  labelInvalidIPAddressOrDomainName,
+  labelInvalidURL,
+} from './translatedLabels';
+
+const IPAddressAndDomainRegexp =
+  /^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}(\/\d{1,3})?$|^(\S)+\.{1,}\D+[^.]$/;
+
+const urlRegexp = /https?:\/\/(\S+)/;
```
Comment (jdelpierre) [inline www/front_src/src/Authentication/Openid/useValidationSchema.ts:14]: https, the "s" is mandatory on this feature ?

Comment (Thebarda) [reply www/front_src/src/Authentication/Openid/useValidationSchema.ts:14]: It is not mandatory

**Review by jdelpierre — APPROVED** [review]

**Review by Jordan91 — APPROVED** [review]

**Pull request merged**